### ISSUE TITLE
[Bug]: Highlighting do not reload on changing background

### ISSUE THREAD

**Issue by Shatur** [akinsho/bufferline.nvim#522]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

Highlighting stop reloading on changing background after f248437e31b97f2a91173ee770b181b8d8ccdd42
To reproduce just call `:set background=light` or `:set background=light` on a theme that supports light and dark mode. For example, https://github.com/Shatur/neovim-ayu

### What did you expect to happen?

Highlighting changes on changing background setting.

### Config

No configuration required.

### commit

f248437e31b97f2a91173ee770b181b8d8ccdd42

**Comment by akinsho** [comment]
@Shatur I just tested this out with my colorscheme `doom-one.nvim` and setting background light/dark works without issue. Is this problem specific to your theme or can you reproduce it for other colorschemes

**Comment by Shatur** [comment]
Strange, I can reproduce it with https://github.com/NTBBloodbath/doom-one.nvim too. Even without configuration:

![изображение](https://user-images.githubusercontent.com/22453358/186439796-348decc5-8110-4b1d-a6f0-7ea44cb03f08.png)
![изображение](https://user-images.githubusercontent.com/22453358/186439778-9ec97097-1fff-4adf-9713-18a40b1d8204.png)

I use NVIM v0.7.2

**Comment by akinsho** [comment]
@Shatur can you reproduce this with a minimal `init.lua` that just packadds this plugin and doom and sets the colorscheme then loads this plugin with no config. Maybe it has to do with the fact that I am on nvim nightly but nothing I changed was nightly specific but the behaviour might be different somehow

**Comment by akinsho** [comment]
Okay I'm able to reproduce it with
```lua
vim.cmd.packadd("local-bufferline.nvim")
vim.cmd.packadd("doom-one.nvim")

vim.opt.termguicolors = true

vim.cmd.colorscheme("doom-one")

require("bufferline").setup()
```

I'll look at it when I have some free time, would be good to get some sort of test for this since colorscheme relate breakage is constant but very hard to test

**Comment by Shatur** [comment]
> Okay I'm able to reproduce it with

Yes, I debugged it and it looks like the issue is caused by calling `require("bufferline").setup()` after `vim.cmd.colorscheme("doom-one")`.
Here is minimal config to reproduce the issue (run with `nvim -nu minimal.lua`):
```lua
vim.cmd [[set runtimepath=$VIMRUNTIME]]
vim.cmd [[set packpath=/tmp/nvim/site]]
local package_root = '/tmp/nvim/site/pack'
local install_path = package_root .. '/packer/start/packer.nvim'
local function load_plugins()
  require('packer').startup {
    {
      'wbthomason/packer.nvim',
      'akinsho/bufferline.nvim',
      'NTBBloodbath/doom-one.nvim',
    },
    config = {
      package_root = package_root,
      compile_path = install_path .. '/plugin/packer_compiled.lua',
      display = { non_interactive = true },
    },
  }Take a loo
end
_G.load_config = function()
  vim.o.termguicolors = true
  vim.cmd("colorscheme doom-one")
  require('bufferline').setup({})
end
if vim.fn.isdirectory(install_path) == 0 then
  print("Installing bufferline.nvim and dependencies.")
  vim.fn.system { 'git', 'clone', '--depth=1', 'https://github.com/wbthomason/packer.nvim', install_path }
end
load_plugins()
require('packer').sync()
vim.cmd [[autocmd User PackerComplete ++once echo "Ready!" | lua load_config()]]
```

I took it from https://github.com/nvim-telescope/telescope.nvim/blob/master/.github/ISSUE_TEMPLATE/bug_report.yml

**Comment by akinsho** [comment]
fixed by #523
